File: VB/NetFramework/net48/13_Touch/TouchToolBar.vb
Imports System.Windows.Forms
Imports GrapeCity.Win.Editors

Namespace _13_Touch
    Partial Public Class TouchToolBar
        Inherits InputManWin13_Demo.DemoBase_settings

        Public Sub New()
            InitializeComponent()
            InitializeForm()
        End Sub

        Private Sub InitializeForm()
            ' xmlファイルからデータ取得(GcComboBox)
            Dim filePath As System.String = Application.StartupPath
            Me.dataSet1.ReadXml(filePath & "\Resources\prefecture.xml")
            gcComboBox1.DataSource = dataSet1.Tables(0)
            gcComboBox1.SelectedIndex = -1

            For i As Integer = 1 To gcComboBox1.ListColumns.Count - 1
                gcComboBox1.ListColumns(i).AutoWidth = True
            Next

            ' xmlファイルからデータ取得(GcMaskedComboBox)
            Me.DataSet2.ReadXml(filePath & "\Resources\pref.xml")
            GcMaskedComboBox1.DataSource = DataSet2.Tables(0)
            GcMaskedComboBox1.SelectedIndex = -1

            For i As Integer = 1 To GcMaskedComboBox1.ListColumns.Count - 1
                GcMaskedComboBox1.ListColumns(i).AutoWidth = True
            Next
        End Sub

        Private Sub checkBox_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles checkBox1.CheckedChanged, checkBox3.CheckedChanged, checkBox2.CheckedChanged

            Dim DisplayOption As Integer = 0

            If checkBox1.Checked Then
                DisplayOption = DisplayOption Or TouchToolBarDisplayOptions.PressAndHold
            End If
            If checkBox2.Checked Then
                DisplayOption = DisplayOption Or TouchToolBarDisplayOptions.TapSelection
            End If
            If checkBox3.Checked Then
                DisplayOption = DisplayOption Or TouchToolBarDisplayOptions.TapGripper
            End If

            gcTextBox1.ShowTouchToolBar = CType(DisplayOption, TouchToolBarDisplayOptions)
            gcMask1.ShowTouchToolBar = CType(DisplayOption, TouchToolBarDisplayOptions)
            gcCharMask1.ShowTouchToolBar = CType(DisplayOption, TouchToolBarDisplayOptions)
            gcDateTime1.ShowTouchToolBar = CType(DisplayOption, TouchToolBarDisplayOptions)
            gcDate1.ShowTouchToolBar = CType(DisplayOption, TouchToolBarDisplayOptions)
            gcTime1.ShowTouchToolBar = CType(DisplayOption, TouchToolBarDisplayOptions)
            gcTimeSpan1.ShowTouchToolBar = CType(DisplayOption, TouchToolBarDisplayOptions)
            gcNumber1.ShowTouchToolBar = CType(DisplayOption, TouchToolBarDisplayOptions)
            gcComboBox1.ShowTouchToolBar = CType(DisplayOption, TouchToolBarDisplayOptions)
            GcMaskedComboBox1.ShowTouchToolBar = CType(DisplayOption, TouchToolBarDisplayOptions)
        End Sub
    End Class
End Namespace

File: CS/NetFramework/net48/13_Touch/TouchToolBar.cs
using System;
using System.Windows.Forms;
using GrapeCity.Win.Editors;

namespace InputManWin13_Demo._13_Touch
{
    public partial class TouchToolBar : InputManWin13_Demo.DemoBase_settings
    {
        public TouchToolBar()
        {
            InitializeComponent();
            InitializeForm();
        }

        private void InitializeForm()
        {
            // ***** 初期設定
            checkBox1.Click += new EventHandler(checkBox_Click);
            checkBox2.Click += new EventHandler(checkBox_Click);
            checkBox3.Click += new EventHandler(checkBox_Click);

            // xmlファイルからデータ取得
            System.String filePath = Application.StartupPath;
            this.dataSet1.ReadXml(filePath + "\\Resources\\prefecture.xml");
            gcComboBox1.DataSource = dataSet1.Tables[0];
            gcComboBox1.SelectedIndex = -1;

            for (int i = 1; i < gcComboBox1.ListColumns.Count; i++)
            {
                gcComboBox1.ListColumns[i].AutoWidth = true;
            }

            // xmlファイルからデータ取得（gcMaskedComboBox）
            this.dataSet2.ReadXml(filePath + "\\Resources\\pref.xml");
            gcMaskedComboBox1.DataSource = dataSet2.Tables[0];
            gcMaskedComboBox1.SelectedIndex = -1;

            for (int i = 1; i < gcMaskedComboBox1.ListColumns.Count; i++)
            {
                gcMaskedComboBox1.ListColumns[i].AutoWidth = true;
            }

        }

        void checkBox_Click(object sender, EventArgs e)
        {
            int option = 0;

            if (checkBox1.Checked)
            {
                option = option | (int)TouchToolBarDisplayOptions.PressAndHold;
            }
            if (checkBox2.Checked)
            {
                option = option | (int)TouchToolBarDisplayOptions.TapSelection;
            }
            if (checkBox3.Checked)
            {
                option = option | (int)TouchToolBarDisplayOptions.TapGripper;
            }

            gcTextBox1.ShowTouchToolBar = (TouchToolBarDisplayOptions)option;
            gcMask1.ShowTouchToolBar = (TouchToolBarDisplayOptions)option;
            gcCharMask1.ShowTouchToolBar = (TouchToolBarDisplayOptions)option;
            gcDateTime1.ShowTouchToolBar = (TouchToolBarDisplayOptions)option;
            gcDate1.ShowTouchToolBar = (TouchToolBarDisplayOptions)option;
            gcTime1.ShowTouchToolBar = (TouchToolBarDisplayOptions)option;
            gcTimeSpan1.ShowTouchToolBar = (TouchToolBarDisplayOptions)option;
            gcNumber1.ShowTouchToolBar = (TouchToolBarDisplayOptions)option;
            gcComboBox1.ShowTouchToolBar = (TouchToolBarDisplayOptions)option;
            gcMaskedComboBox1.ShowTouchToolBar = (TouchToolBarDisplayOptions)option;
        }
    }
}
